scripts/scil_bundle_meanstdperpoint.py

```diff
@@ -0,0 +1,134 @@
+    p = argparse.ArgumentParser(
+        description='Compute mean and standard deviation for all streamlines '
+                    'points in the bundle for each metric combination',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file to compute statistics on.')
+
+    add_reference(p)
+
+    p.add_argument('label_map',
+                   help='Label map (.npz) of the corresponding '
+                        'fiber bundle.')
+    p.add_argument('distance_map',
+                   help='Distance map (.npz) of the corresponding '
+                        'bundle/centroid streamline.')
+    p.add_argument('metrics', nargs='+',
+                   help='Nifti metric(s) to compute statistics on.')
+    p.add_argument('--density_weighting', action='store_true',
+                   help='If set, weight statistics by the number of '
+                        'tracks passing through each voxel.')
+    p.add_argument('--distance_weighting', action='store_true',
+                   help='If set, weight statistics by the inverse of the '
+                        'distance between a streamline and the centroid.')
+
+    add_json_args(p)
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_bundle, args.label_map,
+                                 args.distance_map] + args.metrics)
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+    sft.to_vox()
+    sft.to_corner()
+
+    stats = {}
+    bundle_name, _ = os.path.splitext(os.path.basename(args.in_bundle))
+    if len(sft.streamlines) == 0:
+        stats[bundle_name] = None
+        print(json.dumps(stats, indent=args.indent, sort_keys=args.sort_keys))
+        return
+
+    metrics = [nib.load(m) for m in args.metrics]
+    assert_same_resolution(*metrics)
+
+    if args.density_weighting:
+        track_count = compute_tract_counts_map(
+            sft.streamlines, metrics[0].shape).astype(np.float64)
+    else:
+        track_count = np.ones(metrics[0].shape)
+
+    label_file = np.load(args.label_map)
+    labels = label_file['arr_0']
+
+    unique_labels = np.unique(labels)
+    num_digits_labels = len(str(np.max(unique_labels)))
+
+    distance_file = np.load(args.distance_map)
+    distances_to_centroid_streamline = distance_file['arr_0']
+    # Bigger weight near the centroid streamline
+    distances_to_centroid_streamline = 1.0 / distances_to_centroid_streamline
+
+    if len(labels) != len(distances_to_centroid_streamline):
+        raise Exception(
+            'Label map doesn\'t contain the same number of '
+            'entries as the distance map. {} != {}'
+            .format(len(labels), len(distances_to_centroid_streamline)))
+
+    bundle_data_int = sft.streamlines.data.astype(np.int)
+    stats[bundle_name] = {}
+    for metric in metrics:
+        metric_data = metric.get_data()
```

Comment:
When we will switch to Dipy 1.1 or 1.2, we should also switch to Nibabel 3.0 and switch to `get_fdata()` and when expecting `int` casting it ourselves.